chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -420,197 +420,206 @@
-                target_platform_version: target_platform_version)
-    end
+      target_platform_version = os.gsub(/^.*-/, "")
+      template "#{os_dir}/crowbar_join.sh" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "crowbar_join.suse.sh.erb"
+        variables(admin_ip: admin_ip,
+                  web_port: web_port,
+                  ntp_servers_ips: ntp_servers_ips,
+                  target_platform_version: target_platform_version)
+      end
 
-    repos = Provisioner::Repositories.get_repos("suse", target_platform_version)
+      repos = Provisioner::Repositories.get_repos("suse", target_platform_version, arch)
 
-    packages = node[:provisioner][:packages][os] || []
+      packages = node[:provisioner][:packages][os] || []
 
-    template "#{os_dir}/crowbar_register" do
-      mode 0644
-      owner "root"
-      group "root"
-      source "crowbar_register.erb"
-      variables(admin_ip: admin_ip,
-                admin_broadcast: admin_broadcast,
-                web_port: web_port,
-                ntp_servers_ips: ntp_servers_ips,
-                os: os,
-                crowbar_key: crowbar_key,
-                domain: domain_name,
-                repos: repos,
-                packages: packages,
-                target_platform_version: target_platform_version)
-    end
+      template "#{os_dir}/crowbar_register" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "crowbar_register.erb"
+        variables(admin_ip: admin_ip,
+                  admin_broadcast: admin_broadcast,
+                  web_port: web_port,
+                  ntp_servers_ips: ntp_servers_ips,
+                  os: os,
+                  arch: arch,
+                  crowbar_key: crowbar_key,
+                  domain: domain_name,
+                  repos: repos,
+                  packages: packages,
+                  target_platform_version: target_platform_version)
+      end
 
-    missing_files = ! File.exists?("#{os_dir}/install/boot/x86_64/common")
+      missing_files = !File.exist?("#{os_dir}/install/boot/#{arch}/common")
 
-  when /^(redhat|centos)/ =~ os
-    # Add base OS install repo for redhat/centos
-    if ::File.exists? "#{tftproot}/#{os}/install/repodata"
-      node.set[:provisioner][:repositories][os]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install" => true }
-    else
-      node.set[:provisioner][:repositories][os]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install/Server" => true }
-    end
-    # Default kickstarts and crowbar_join scripts for redhat.
+    when /^(redhat|centos)/ =~ os
+      # Add base OS install repo for redhat/centos
+      if ::File.exist? "#{tftproot}/#{os}/#{arch}/install/repodata"
+        node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{admin_web}" => true }
+      else
+        node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{admin_web}/Server" => true }
```

Comment:
Metrics/LineLength: Line is too long. [107/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)